docs/NEW_GUIDE_DRAFT.en.md

```diff
@@ -0,0 +1,68 @@
+`NEAR CLI Core commands`
+login
+create-account
+    - implicit
+    - subaccount
+<!-- account Id is entered in this step -->
+manage-account
+    - keys
+        - view
+        - add
+        - delete
+        - nonce
+    - local-keys
+        <!-- Needs to be designed (with multikey management) -->
+    - contract
+        - deploy
+        - dev-deploy
+        - state
+        - wasm-hash
+        - view-call
+        - change-call
+    - state
+    - delete
+transfer
+<!-- TODO: feels a bit out of place, but definitely important. Maybe we need to add other view functions and have 1 subcommand for them. Or it can be one of the helpers-->
+transaction-status
+helpers
+    <!-- please, suggest other helpers -->
+    - generate key
+	- get the public key from ledger device
+    - recent-block-hash
+
+<!-- Currently it's a part of NEAR CLI Rust. Seems like a functionality for advanced users -->
+`transaction-constructor` extension
+    - constract-transaction
+    - sign a transaction with the private key
+	- combine unsigned transaction with a signature
+	- sign a transaction with ledger
+	- send signed transaction
+	- deserializing the bytes from base64
+
+<!-- Most of the people will not use staking functionality from CLI, lets move it to extension. -->
+`staking` extension
+    <!-- TODO: what is this command? -->
+    - new-stake-proposal
+    - stake
+    <!-- NOTE: new command, should it be here? -->
+    - unstake
+
+NEAR CLI is built for:
+- NEAR smart-contract developers
+- validators
+- ..?
+
+Mental model to distinguish extension from the core commands:
+- `NEAR CLI core` is a single binary and contains all of the `core` commands
+- Each extension is a separate binary that can be executed from `NEAR CLI`
+- There is no extensions that are installed `by default`
+- `NEAR CLI` Core commands should be usefull for all groups of users
+- Extensions are used by a particular group or several groups of users
+- Extensions are not composable
+
+Open questions:
+- where to add flags like --verbose, --structured, etc.
```

Comment:
In my head, `--verbose` is for errors, and `--structured` will be used to return structured results, for example, JSON.
Example:
```bash
near account serhii.near balance
```
Result:
```bash
10.1 N
```
```bash
near account serhii.near balance --structured
```
Result:
```bash
{
    "balance": "234567890987654",
    "locked": "7654567654",
    ...
}
```

Reply:
I see. Well `--structured` is not the best naming since it is not intuitive and also does not communicate explicitly that the output will be in JSON. I don't know any CLI tools that I could get inspiration from for this type of behavior, but I feel it should be defined on the top-level together with `verbose` flag.

I believe we can leave it as an open question, but I would consider to make `--structured` to be a parameter instead of a flag, thus one can potentially choose `human-readable` vs `json` vs `csv` vs ... formats.

Reply:
Ok, we can use `--json` and extend it with other formats if needed. Probably `human readable` should be by default. Top-level flag should be the best approach, I believe.
If we will make it a parameter, people will have to choose one of the options every time. I'm not sure is we should complicate things here.

Reply:
But yes, I will leave it in open questions for now. Closing this discussion for now.